### PR TITLE
Changes related to Symmetric Hessian

### PR DESCRIPTION
**Pull request by abelsiqueira**
Related to https://github.com/JuliaSmoothOptimizers/NLPModels.jl/pull/354

### CODE REVIEW

**Comment by abelsiqueira** [comment]
This has been rebased and updated with a bump on 0.17, which doesn't affect anything. 0.16 is breaking so the version has been updated to 0.3.0.

**Comment by abelsiqueira** [comment]
It just started failing: https://github.com/JuliaDiff/ForwardDiff.jl/issues/529.

**Comment by abelsiqueira** [comment]
It's fixed now.

**Comment by geoffroyleconte** [comment]
Hi, thanks for this PR! Can we merge it if everything is fixed? Some tests failed but I don't know if they were run prior to the ForwardDiff modifications

File: test/nlp/nlpmodelstest.jl [hunk 1]
```diff
@@ -1,6 +1,6 @@
 
 @testset "AD backend - $(adbackend)" for adbackend in (:ForwardDiffAD, :ZygoteAD, :ReverseDiffAD)
-  for problem in NLPModelsTest.nlp_problems
+  for problem in NLPModelsTest.nlp_problems[1:1]
```
Comment (abelsiqueira) [inline test/nlp/nlpmodelstest.jl:3]:
```suggestion
  for problem in NLPModelsTest.nlp_problems
```

**Comment by abelsiqueira** [comment]
Someone needs to review. I asked for @dpo, but @tmigot or @amontoison are both welcome as well.

**Review by amontoison — APPROVED** [review]

**Comment by abelsiqueira** [comment]
Thanks

**Comment by geoffroyleconte** [comment]
Thanks a lot!

**Pull request merged**